docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -0,0 +1,1378 @@
+
+_The row order in this table is important.  Type lookup stops after first match is found scanning from top to bottom (with the single exception of "null" - first row in the table)_
+<Tabs groupId="type-casting">
+  <TabItem value="redshift" label="Redshift and Postgres" default>
+
+<table>
+<thead>
+<td>Json Schema</td>
+<td>Redshift/Postgres Type</td>
+</thead>
+<tbody>
+<tr>
+<td >
+
+  ```json
+{
+    "type": ["null", T1, ...]
+}
+```
+
+OR
+
+  ```json
+{
+    "enum": ["null", E1, ...]
+}
+```
+
+`"null"` is not considered for type casting logic. Only for nullability constraint. Type lookup will continue down the table.
+
+</td>
+<td>
+
+`NULLABLE`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "enum": [E1, E2, ...]
+}
+```
+
+-  `M` is maximum size of `json.stringify(E*)`
+
+
+The `enum` can contain more than **one** JavaScript type: `string`, `number|integer`, `boolean`.
+For the purposes of this  `number` and `integer` are the same.
+
+
+`array`, `object`, `NaN` and other types in enum will be cast as fallback `VARCHAR(65535)`.
+
+</td>
+<td>
+
+`VARCHAR(M)`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": [T1, T2, ...]
+}
+```
+
+- `T1, T2, ..`. only contains `"boolean"` **and** `"integer"`
+
+</td>
+<td>
+
+`VARCHAR(10)`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": [T1, T2, ...]
+}
+```
+
+</td>
+<td>
+
+`VARCHAR(4096)`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "string",
+    "format": "date-time"
+}
+```
+
+</td>
+<td>
+
+`TIMESTAMP`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "string",
+    "format": "date"
+}
+```
+
+</td>
+<td>
+
+`DATE`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "array"
+}
+```
+
+</td>
+<td>
+
+`VARCHAR(65535)`
+
+_Content is strigified and quoted._
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "integer",
+    "maximum": M
+}
+```
+
+- `M` &le; 32767
+
+</td>
+<td>
+
+`SMALLINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "integer",
+    "maximum": M
+}
+```
+
+- 32767 &lt; `M` &le; 2147483647
+
+</td>
+<td>
+
+`INT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "integer",
+    "maximum": M
+}
+```
+
+- `M` &gt;2147483647
+
+</td>
+<td>
+
+`BIGINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "integer",
+    "enum": [E1, E2, ...]
+}
+```
+
+- Maximum `E*` &le; 32767
+
+</td>
+<td>
+
+`SMALLINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "type": "integer",
+    "enum": [E1, E2, ...]
+}
+```
+
+- 32767 &lt; maximum `E*` &le; 2147483647
+
+</td>
+<td>
+
+`INT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "integer",
+    "enum": [E1, E2, ...]
+}
+```
+
+- Maximum `E*` &gt; 2147483647
+
+</td>
+<td>
+
+`BIGINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "integer"
+}
+```
+
+</td>
+<td>
+
+`BIGINT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "multipleOf": B
+}
+```
+
+</td>
+<td>
+
+`INT`
+
+</td>
+</tr>
+<tr>
+<td>
+
+```json
+{
+    "type": "number",
+    "multipleOf": B
```

Comment:
Should we just say `"multipleOf": 2`? Or that’s confusing?

Reply:
I'd say yes. Readers might thing that they can generalize 2 to any number.